### ISSUE TITLE
Linter: configuration for specific rules (e.g. eqeqeq)

### ISSUE THREAD

**Issue by LinusU** [oxc-project/oxc#1757]
Sorry if this information is available somewhere, I've searched around the site and GitHub quite extensively...

I'm looking at adding Oxlint to our workflow, and hopefully replace ESLint in the long run. I found it a bit hard to understad which rules are currently supported but I have now understood that `crates/oxc_linter/src/rules/eslint` has files corresponding to the ESLint rule names 🎉 

Trying to enable the same rules as we currently use in ESLint led to a hiccup though. My approach was something like: run `eslint --print-config rules` and keep the supported rules in a file called `oxlint.json`. Then run Oxlint with this config, `oxlint --config oxlint.json`.

However, this quickly led to a number of failed lints with the `eqeqeq` rule, since we are using it with the config `"null": "ignore"`.

This makes it so that triple equals is enforced for everything but `null` checks. I believe that this is a very good way to handle `===`, and when I see `=== undefined` or `=== null` I always look extra closely at the code as I've seen this introduce bugs numerous times.

For those not familiar, `x == null` is the equivalent of `x === null || x === undefined`. With how null is used in the wider JavaScript ecosystem, I've found that these cases are very common:

1. Libraries with functions that are documented to return `undefined`, but actually return `null`
2. Libraries with functions that are documented to return `null`, but actually return `undefined`
3. Libraries with functions that are documented to return `null`, but in some cases return `undefined` (e.g. due to early return without `null`)
4. Libraries with functions that change between `null`/`undefined` in a non-major versioned release

Because of this, I've taken to be liberal when receiving `null` or `undefined`, and always use `== null` unless there's supposed to be a clear distinction of `null` vs `undefined` (e.g. in a patch `null` could be remove this field, and `undefined` would be keep as is).

-----

So, that was a lot of background to ask: Is configuration of rules something that is on the radar? (or perhaps already possible, just missing in eqeqeq?)

**Comment by LinusU** [comment]
Here is a failing test case:

**`test.js`**

```js
const foo = (Math.random() * 3) | 0
const bar = [undefined, null, 'test'][foo]

if (bar != null) {
  console.log(bar)
} else {
  console.log('bar is undefined or null')
}
```

**`oxlint.json`**

```json
{
  "rules": {
    "eqeqeq": ["error", "always", { "null": "ignore" }]
  }
}
```

**command**

```text
$ npx oxlint@latest --config oxlint.json

  ⚠ eslint(eqeqeq): Expected !== and instead saw !=
   ╭─[a.js:3:1]
 3 │ 
 4 │ if (bar != null) {
   ·     ───────────
 5 │   console.log(foo)
   ╰────
  help: Prefer !== operator

Finished in 8ms on 1 file with 1 rules using 8 threads.
Found 1 warning and 0 errors.
```

**Comment by mysteryven** [comment]
Oh, we supported read ESLint rule config below now, we also need to read the third part `{"null": "ignore"}` 

```js
"rule": "off",
"rule": ["off", "config"],
```
